### PR TITLE
V2 beta fix open parse

### PR DESCRIPTION
**Pull request by dteach-rv**
Worker.cpp Bug Fixes and Logging Updates.

 - Assertion errors in the Worker::work sub-routine were being caused by a corner case
   where, the last message in the buffer was able to be parsed correctly, thus not
   triggering a PARTIAL_MSG error, but still long enough to overflow the
   bmp_data_unread_len counter. The simple fix is to trigger a buffer refill once the
   data_unread_len approaches 0. I've arbitrarily set this to 64. A more efficient
   solution could take into account the bmp_raw_msg_len subtracted from the unread_len
   and trigger only when an overflow would occur.
 - Additional logging cases were added to provide a more complete log output for debug.
 - A buffer overflow scenario is now being logged in the refill_buffer method.
 - Additional logging was added before the assertions in update_buffer.

### CODE REVIEW

File: src/Worker.cpp [hunk 1]
```diff
@@ -207,19 +237,28 @@ void Worker::refill_buffer(int recv_len) {
...
 void Worker::update_buffer(int parsed_bmp_msg_len) {
     bmp_data_read_len += parsed_bmp_msg_len;
     bmp_data_unread_len -= parsed_bmp_msg_len;
-    assert(bmp_data_read_len >= 0);
+    if(bmp_data_read_len < 0){
+        LOG_ERR("Assertion error. bmp_data_read_len is invalid (%d)", bmp_data_read_len);
+    }
+    assert(bmp_data_unread_len >= 0);
+    if(bmp_data_unread_len < 0){
+        LOG_ERR("Assertion error. bmp_data_unread_len is invalid (%d)", bmp_data_read_len);
```
Comment (ryeleo) [inline src/Worker.cpp]: Copy paste typo -- want to print `unread`

File: src/Worker.cpp [hunk 2]
```diff
@@ -229,5 +268,4 @@ int Worker::get_bmp_data_unread_len() {
 
 uint8_t *Worker::get_unread_buffer() {
     return bmp_data_buffer + bmp_data_read_len;
-}
-
+}
```
Comment (ryeleo) [inline src/Worker.cpp]: Newlines at end of file are kind for cli users!

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,5 @@
 cmake-get_encap_msg*
 cmake-build*
 get_encap_msg
-
+/build/
+/.vscode/
```
Comment (ryeleo) [inline .gitignore]: Newlines at end of file are kind for cli users!

**Review by ryeleo — CHANGES_REQUESTED** [review]
Just fix the typo in the log message. Otherwise, looks great! 👍👍

File: src/Worker.cpp (hunk 1 above)
Comment (dteach-rv) [reply src/Worker.cpp]: Fixed.

File: src/Worker.cpp (hunk 2 above)
Comment (dteach-rv) [reply src/Worker.cpp]: Fixed.

File: .gitignore (hunk 3 above)
Comment (dteach-rv) [reply .gitignore]: Fixed.

**Pull request merged**